### PR TITLE
python3Packages.aiorecollect: init 1.0.2

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
aiorecollect is a Python asyncio-based library for the ReCollect
Waste API. It allows users to programmatically retrieve schedules
for waste removal in their area, including trash, recycling, compost
and more.

https://github.com/bachya/aiorecollect

This is a Home Assistant dependency.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 112199` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.aiorecollect</li>
    <li>python38Packages.aiorecollect</li>
    <li>python39Packages.aiorecollect</li>
  </ul>
</details>

File: pkgs/development/python-modules/aiorecollect/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    pytest-cov
+    pytestCheckHook
+  ];
+
+  # Ignore the examples as they are prefixed with test_
+  pytestFlagsArray = [ "--ignore examples/" ];
+  pythonImportsCheck = [ "aiorecollect" ];
+
+  meta = with lib; {
+    description = " Python library for the Recollect Waste API";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/aiorecollect/default.nix:43]:
```suggestion
    description = "Python library for the Recollect Waste API";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112199` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.aiorecollect</li>
<li>python38Packages.aiorecollect</li>
<li>python39Packages.aiorecollect</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112199` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.aiorecollect</li>
<li>python38Packages.aiorecollect</li>
<li>python39Packages.aiorecollect</li>
</ul>
</details>

File: pkgs/development/python-modules/aiorecollect/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/aiorecollect/default.nix]: Thanks, removed.

File: pkgs/development/python-modules/aiorecollect/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  checkInputs = [
+    aresponses
+    freezegun
+    pytest-asyncio
+    pytest-cov
+    pytestCheckHook
+  ];
+
+  # Ignore the examples as they are prefixed with test_
+  pytestFlagsArray = [ "--ignore examples/" ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/aiorecollect/default.nix:39]:
```suggestion
  disabledTestFiles = [ "examples" ];
```

This just got merged recently.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/aiorecollect/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/development/python-modules/aiorecollect/default.nix:39]: Doesn't work in the case here. `examples` is a directory and located in the project's root.

```bash
Executing pytestCheckPhase
Disabled test file "examples" does not exist. Aborting
```

**Pull request merged**